lms/static/scripts/frontend_apps/components/LMSFilePicker.js

```diff
@@ -92,11 +93,47 @@ export default function LMSFilePicker({
   // Authorization attempt was made. Set after state transitions to "authorizing".
   const [authorizationAttempted, setAuthorizationAttempted] = useState(false);
 
-  // The file within `files` which is currently selected.
-  const [selectedFile, selectFile] = useState(/** @type {File|null} */ (null));
+  // The file or folder within `files` which is currently selected.
+  const [selectedFile, setSelectedFile] = useState(
+    /** @type {File|null} */ (null)
+  );
+
+  // An array of File objects representing the  "path" to the current
+  // list of files being displayed. This always starts at the root. The last
+  // element represents the current directory path.
+  const [folderPath, setFolderPath] = useState(
+    /** @type File[] */ ([
+      {
+        display_name: 'Files',
+        type: 'Folder',
+        contents: listFilesApi,
+        id: '__root__',
+      },
+    ])
+  );
```

Comment:
I experimented with having `LMSFilePicker` be only aware of the current path, and make `Breadcrumbs` own the path state overall, but it didn't feel right. This "manufacturing" of a `File` object out of the `listFilesApi` may be somewhat inelegant, but it is nice to be able to build an Array of `File` objects.